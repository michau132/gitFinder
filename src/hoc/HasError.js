import React from 'react';
import PropTypes from 'prop-types';
import { Typography } from '@material-ui/core';

const HasError = Component => (props) => {
  const { restStore: { error } } = props;
  if (error) {
    return (
      <Typography variant="h5" color="error">
        Ups something goes wrong or user does not exist
      </Typography>
    );
  }
  return <Component {...props} />;
};

HasError.propTypes = {
  restStore: PropTypes.shape({
    error: PropTypes.bool.isRequired,
  }).isRequired,
};

export default HasError;
