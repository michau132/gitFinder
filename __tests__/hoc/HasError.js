import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import HasError from '../../src/hoc/HasError';

const component = () => <h1>Hi there</h1>;
const ConditionalHOC = HasError(component);
const props = {
  restStore: { error: true },
};

describe('testing HasError HOC', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ConditionalHOC {...props} />);
  });
  test('should render component when loading is set to true', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
