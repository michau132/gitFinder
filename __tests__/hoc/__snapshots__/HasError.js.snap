// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing HasError HOC should match snapshot 1`] = `
<WithStyles(Typography)
  color="error"
  variant="h5"
>
  Ups something goes wrong or user does not exist
</WithStyles(Typography)>
`;
